### PR TITLE
2171: Implementing Campaign Option and Campaign Preset Versioning

### LINKED ISSUE #2171: Implement Campaign Option Preset Versioning
This issue calls for the implementation of proper versioning for Campaign Options presets.

### PR DESCRIPTION
**Pull request by Windchild292**
This fixes #2171

### CODE REVIEW

File: MekHQ/src/mekhq/Version.java [hunk 1]
```diff
@@ -190,4 +190,10 @@ private static int compare(Version left, Version right) {
 
         return 0;
     }
+
+    @Override
+    public String toString() {
+        return hasRevision() ? String.format("%d-%d-%d-%d", getMajorVersion(), getMinorVersion(), getSnapshot(), getRevision())
+                : String.format("%d-%d-%d", getMajorVersion(), getMinorVersion(), getSnapshot());
```
Comment (sixlettervariables) [inline MekHQ/src/mekhq/Version.java:197]: Why dashes instead of dots?

**Review by sixlettervariables — COMMENTED** [review]
One question regarding toString.

File: MekHQ/src/mekhq/Version.java (hunk 1 above)
Comment (Windchild292) [reply MekHQ/src/mekhq/Version.java:197]: Airhead mistake after looking at dates

Comment (Windchild292) [reply MekHQ/src/mekhq/Version.java]: Fixed and pushed.

**Review by sixlettervariables — APPROVED** [review]

**Pull request merged**